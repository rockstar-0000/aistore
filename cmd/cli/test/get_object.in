echo "0123456789" > $OBJECT_1.txt
ais bucket create ais://$BUCKET_1 | awk '{print $1 " " $2}'
ais object put $OBJECT_1.txt ais://$BUCKET_1/object_1.txt
rm -f $OBJECT_1.txt

# Negative Testing
ais object get ais://$BUCKET_2/object_1.txt $OBJECT_1.txt // FAIL "bucket "ais://$BUCKET_2" does not exist"
ais object get ais://$BUCKET_1/object_2.txt $OBJECT_1.txt // FAIL ""ais://$BUCKET_1/object_2.txt" does not exist"
ais object get aws://$BUCKET_1/object_1.txt $OBJECT_1.txt // FAIL

# Positive Testing
ais object get ais://$BUCKET_1/object_1.txt $OBJECT_1.txt
$PRINT_SIZE $OBJECT_1.txt
ais object get ais://$BUCKET_1/object_1.txt $OBJECT_2.txt
$PRINT_SIZE $OBJECT_2.txt
rm -f $OBJECT_*

# Test with no OUT_FILE argument specified
ais object get ais://$BUCKET_1/object_1.txt
$PRINT_SIZE object_1.txt
rm -f object_1.txt

# Set up file and directory structure for the next tests
for d1 in {0..2}; do for d2 in {0..2}; do mkdir -p /tmp/rdir/test$d1/$d2 && echo -n "$d1 $d2" > /tmp/rdir/test$d1/$d2/test.txt; done; done
ais object put "/tmp/rdir/test{0..2}/{0..2}/test.txt" ais://$BUCKET_1 -y // IGNORE
RES=0 && for d1 in {0..2}; do for d2 in {0..2}; do ais object get ais://$BUCKET_1/test$d1/$d2/test.txt && RES=$((RES+1)); done; done && echo $RES

# Test get to stdout
ais object get ais://$BUCKET_1/test2/2/test.txt -

# Expect the last retrieved object to overwrite previous gets of the same file name
cat test.txt

rm test.txt // IGNORE
rm -rf /tmp/rdir // IGNORE
