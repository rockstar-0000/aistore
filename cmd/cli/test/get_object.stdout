"ais://$BUCKET_1" created
^PUT.*=> ais://$BUCKET_1/object_1.txt$

^GET object_1.txt from ais://$BUCKET_1.*$
11
^GET object_1.txt from ais://$BUCKET_1.*$
11
^GET object_1.txt from ais://$BUCKET_1.*$
11
^GET test0/0/test.txt from ais://$BUCKET_1.*$
^GET test0/1/test.txt from ais://$BUCKET_1.*$
^GET test0/2/test.txt from ais://$BUCKET_1.*$
^GET test1/0/test.txt from ais://$BUCKET_1.*$
^GET test1/1/test.txt from ais://$BUCKET_1.*$
^GET test1/2/test.txt from ais://$BUCKET_1.*$
^GET test2/0/test.txt from ais://$BUCKET_1.*$
^GET test2/1/test.txt from ais://$BUCKET_1.*$
^GET test2/2/test.txt from ais://$BUCKET_1.*$
9
2 2
2 2
